### PR TITLE
Handle Windows Agent Version

### PR DESCRIPTION
**Pull request by pyrodie18**
##### SUMMARY
Provides a fix to #990 by setting a default value specifically for Windows, and then ignoring other sanity checks.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- Agent Role

### CODE REVIEW

**Review by BGmot — CHANGES_REQUESTED** [review]
Please add this variable to roles/zabbix_agent/vars/Windows.yml

**Comment by pyrodie18** [comment]
There you go @BGmot

**Review by BGmot — COMMENTED** [review]
I am sorry, why do we need to pay special attention to WIndows and even introduce new variable `zabbix_agent_windows_version`, why can't we do in vars/Windows.yml exactly the same as in vars/Debian.yml so no need to modify tasks/main.yml at all. What am I missing here?

**Comment by pyrodie18** [comment]
Honest answer is that I have no idea what comes out of `{{ ansible_distribution_major_version }}` for Windows and I don't have a box to test it on.

**Comment by BGmot** [comment]
> Honest answer is that I have no idea what comes out of `{{ ansible_distribution_major_version }}` for Windows and I don't have a box to test it on.

I see. I'll try to test it today.

**Comment by BGmot** [comment]
ansible_distribution_major_version is: 
- '10' for Windows Server 2016
- '10' for Windows Server 2019
- '10' for Windows Server 2022 
I don't have any non-Server versions of Windows available to test -(

**Comment by pyrodie18** [comment]
That'll at least get this started...thanks and updated.

File: roles/zabbix_agent/tasks/main.yml [hunk 1]
```diff
@@ -21,7 +21,9 @@
 - name: Stopping Install of Invalid Version
   fail:
     msg: Zabbix version {{ zabbix_agent_version }} is not supported on {{ ansible_distribution }} {{ ansible_distribution_major_version }}
-  when: not zabbix_valid_version
+  when:
+    - not zabbix_valid_version
+    - ansible_os_family != "Windows"
```
Comment (BGmot) [inline roles/zabbix_agent/tasks/main.yml:26]: Why do we need this additional condition? this should work for all OS's.

Comment (pyrodie18) [reply roles/zabbix_agent/tasks/main.yml:26]: Because I missed that line lol.  One minute

File: roles/zabbix_agent/tasks/main.yml [hunk 2]
```diff
@@ -21,7 +21,8 @@
 - name: Stopping Install of Invalid Version
   fail:
     msg: Zabbix version {{ zabbix_agent_version }} is not supported on {{ ansible_distribution }} {{ ansible_distribution_major_version }}
-  when: not zabbix_valid_version
+  when:
+    - not zabbix_valid_version
```
Comment (BGmot) [inline roles/zabbix_agent/tasks/main.yml:25]:
```suggestion
```
I am sorry to be so picky, but I think we should not touch tasks/main.yml.

File: roles/zabbix_agent/tasks/main.yml [hunk 3]
```diff
@@ -21,7 +21,8 @@
 - name: Stopping Install of Invalid Version
   fail:
     msg: Zabbix version {{ zabbix_agent_version }} is not supported on {{ ansible_distribution }} {{ ansible_distribution_major_version }}
-  when: not zabbix_valid_version
+  when:
```
Comment (BGmot) [inline roles/zabbix_agent/tasks/main.yml:24]:
```suggestion
  when: not zabbix_valid_version
```

**Review by BGmot — APPROVED** [review]

File: roles/zabbix_agent/vars/Windows.yml [hunk 4]
```diff
@@ -1,2 +1,8 @@
 ---
 # vars file for zabbix_agent (Windows)
+zabbix_valid_agent_versions:
+  # Debian
```
Comment (masa-orca) [inline roles/zabbix_agent/vars/Windows.yml:4]: Is it debian? Isn't it windows?

**Pull request merged**